File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.cpp
#include <iostream>
#include <vector>
#include <cmath>
using namespace std;

int solution(vector<vector<int> > land)
{
    int i, j, k;
    int len = land.size();
    
    for(i=1; i<len; ++i) {
        for(j=0; j<4; ++j) {
            int maxVal = 0;
            for(k=0; k<4; ++k) {
                if(j==k) continue;
                maxVal = max(maxVal, land[i-1][k]);
            }
            land[i][j] += maxVal;
        }
    }
    
    int result = 0;
    for(i=0; i<4; ++i) {
        result = max(result, land[len-1][i]);
    }
    return result;
}

File: 프로그래머스/2/12913. 땅따먹기/땅따먹기.py
def solution(land):
    for k in range(1,len(land)):
        for i in range(0,4):
            maxVal = 0
            for j in range(0,4):
                if i == j :
                    continue
                maxVal = max(maxVal, land[k-1][j])
            land[k][i] += maxVal
    result = 0
    for _ in land[len(land)-1]:
        if(result<_):
            result = _
    return result